portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,45 +29,56 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.Arrays;
-import java.util.ArrayList;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns content and comments*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
     ArrayList<String> words = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
-    
+    Query query = new Query("Task");
+
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     
     //iterating over entities within Datastore
     for (Entity entity : results.asIterable()) {
       long id = entity.getKey().getId();
       String title = (String) entity.getProperty("user-comments");
-      long timestamp = (long) entity.getProperty("timestamp");
     }
 
     response.setContentType("application/json;");
-    response.getWriter().println(words);
+
+    String json = "";
+
+    //get max number of comments
+    int numComments = getNumComments(request);
+
+    //convert request number of comments to json
+    if  (words.size() <= numComments){
```

Comment:
For this if statement, I recommend commenting why it's being done (versus what's being done). The "what" can be understood from the code but I don't understand the "why" aspect, so I think it'd be helpful here.